docs/weave/weave-api-spec/03-querying.md

```diff
@@ -0,0 +1,192 @@
+
+## Available bucket paths by package name
+
+- __Username__
+  - [key prefix](https://github.com/iov-one/weave/blob/v0.18.0/cmd/bnsd/x/username/model.go#L73): `username:`
+  - `/usernames` -> takes `id` as data, returns [bnsd/x/username/username.Token](https://github.com/iov-one/weave/blob/master/cmd/bnsd/x/username/codec.proto#L7-L26)
+  - `/usernames?prefix` -> takes prefix as data, returns `ResultSet`
+
+- __Cash__
+  - [key prefix](https://github.com/iov-one/weave/blob/v0.18.0/x/cash/model.go#L18): `cash:`
+  - `/wallets` -> takes `id`(8 bytes), returns [x/cash.Set](https://github.com/iov-one/weave/blob/v0.18.0/x/cash/codec.proto#L11-L14) by id
+  - `/wallets?prefix` -> takes prefix as data, returns `ResultSet`
+
+- __Sigs__
+  - [key prefix](https://github.com/iov-one/weave/blob/v0.18.0/x/sigs/model.go#L117): `sigs:`
+  - `/auth` -> takes address `id`(8 bytes) returns [x/sigs.UserData](https://github.com/iov-one/weave/blob/v0.18.0/spec/proto/x/sigs/codec.proto#L13-L17) by id
+  - `/auth?prefix` -> takes prefix as data, returns `ResultSet`
+
+- __Multisig__
+  - [key prefix](https://github.com/iov-one/weave/blob/v0.18.0/x/multisig/model.go#L86): `multisig:`
+  - `/contracts` -> takes `id`(8 bytes) returns [x/multisig.Contract](https://github.com/iov-one/weave/blob/v0.18.0/x/multisig/codec.proto#L8-L21)
+  - `/contracts?prefix` -> takes prefix as data, returns `ResultSet`
+
+- __Atomic swaps__
+  - [key prefix](https://github.com/iov-one/weave/blob/v0.18.0/x/aswap/model.go#L85): `swap:`
+  - `/aswaps` -> takes `id`(8 bytes) returns [x/aswap.Swap](https://github.com/iov-one/weave/blob/master/x/aswap/codec.proto#L9-L29)
+  - `/aswaps/source` -> takes `source` as secondary index (8 bytes) returns [x/aswap.Swap](https://github.com/iov-one/weave/blob/master/x/aswap/codec.proto#L9-L29)
+  - `/aswaps/destination` -> takes `destination` as secondary index (8 bytes) returns [x/aswap.Swap](https://github.com/iov-one/weave/blob/master/x/aswap/codec.proto#L9-L29)
+  - `/aswaps/preimage_hash` -> takes `preimage_hash` as secondary index (8 bytes) returns [x/aswap.Swap](https://github.com/iov-one/weave/blob/master/x/aswap/codec.proto#L9-L29)
+  - `/aswaps?prefix` -> takes prefix as data, returns `ResultSet`
+
+- __Escrow__
+  - [key prefix](https://github.com/iov-one/weave/blob/v0.18.0/x/escrow/model.go#L113): `escrow:`
+  - `/escrows` -> takes `id`(8 bytes) returns [x/escrow.Escrow](https://github.com/iov-one/weave/blob/master/x/escrow/codec.proto#L9-L28)
+  - `/escrows/source` -> takes `source` as secondary index (8 bytes) returns [x/escrow.Escrow](https://github.com/iov-one/weave/blob/master/x/escrow/codec.proto#L9-L28)
```

Comment:
These are weave.Address -> 20 bytes

Please check all usages above -> ID (sequence) is 8, address is 20